### PR TITLE
Doc cleanup

### PR DESCRIPTION
**Pull request by kordjamshidi**
I just make a pull request to be able to see the changes easier.

### CODE REVIEW

File: docs/Technical API/Sensor/readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Sensor class
+
+`Sensor`s are procedures to access external resources and procedures. For example, reading from raw data, feature engineering staffs, and preprocessing procedures.
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/readme.md:3]: Can we say: Sensors are backbox functions mostly written by the application programmer that can help for pre-processing, feature extraction, and obtaining representations based on external resources, for example getting the lemma of a word or Glove representation of a word.

File: docs/Technical API/Sensor/readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+"Sensors" are looked at as blackboxes in a program. Users use sensors as callable objects. Underlying `forward()` function will be used to calculate an output. One can override `forward()` function to customize how the sensor get the output.
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/readme.md:4]: First thing can you specify any limitations in the input and output of the sensors?

File: docs/Technical API/Sensor/readme.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+Base classes of `Sensor` that in retuen would handle details like caching calcuated results, managing invocation path, and converting input.
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/readme.md:5]: it will be good to describe this from the application programmer first, what is the template that they should use to write their sensors, and what are their constraint?

File: docs/Technical API/Sensor/readme.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    def __call__(
+        self,
+        data_item: Dict[str, Any],
+        force=False,
+        sensor_name="None"
+    )
+```
+
+All sensors are called during training. The call function for the Base Sensor is simple it just tries to evoke the `update_context`. It `update_context` is executed successfully, its output is return. Otherwise, an Error is retured specifying the name of the sensor.
+`__call__` is immidiatly overridden by the `TorchSensor` and the only difference would be to save the `data_item` vaiable in the Class attributes (`self.context_helper`). The only other Sensor that overrides `__call__` is `JointSensor`.
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/readme.md:19]: what is update_context? I did not see this is described before mentioning it.

File: docs/Technical API/Sensor/Class Sensor.md [hunk 5]
```diff
@@ -20,32 +20,104 @@
...
+
+```python
+FunctionalSensor(arg1,arg2,..., forward, label)
+```
+the first few arguments will be inputs from various `concept`s and `property`s. Forward get a python function as input that would parse the inputs. and label is an optional input that would determine wheather or not the output of the `FunationalSensor` is used in loss function.
+
+For examples of this Sensor refer to the following examples:
+
+examples->ACE05->model.py
+
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/Class Sensor.md:38]: you can add a pointer to the actual line of code here that can be clicked on.

File: docs/Technical API/Sensor/Class Sensor.md [hunk 6]
```diff
@@ -20,32 +20,104 @@
...
+examples->SentimentAnalysis->sensors->tweetGraph.py
+
+examples->WIQA->WIQA_aug.py
+
+examples->demo->main.py
+
+examples->squad->BertQA.py
+
 ## 1.1.1 `JointSensor`
+Some sensors will use properties to calculate other properties.`JointSensor` can calculate multiple properties at once.
```
Comment (kordjamshidi) [inline docs/Technical API/Sensor/Class Sensor.md:54]: Maybe the name joint should be changed to something like multisensor?
